src/main/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreams.java

```diff
@@ -529,6 +568,96 @@ private CachingAmazonDynamoDbStreams(AmazonDynamoDBStreams amazonDynamoDbStreams
+                () -> this.loadStreamDescriptionForAllShards(request));
+            return result;
+        } catch (UncheckedExecutionException | ExecutionException | InvalidCacheLoadException e) {
+            // Catch exceptions thrown on cache lookup or cache loader and try load method once more.
+            // (get call will throw UncheckedExecutionException before aws exception (i.e. AmazonDynamoDBException)).
+            LOG.warn("Failed getting DescribeStreamResult for all shards from cache lookup or load method.  "
+                + "Retrying describeStream call.");
+            e.printStackTrace();
+            return this.loadStreamDescriptionForAllShards(request);
+        }
+    }
+
+    /**
+     * Gets the {@code DescribeStreamResult} from the DescribeStream API.
+     *
+     * @return Stream details for the given request with all the shards currently available.
+     */
+    protected DescribeStreamResult loadStreamDescriptionForAllShards(DescribeStreamRequest request)
+        throws AmazonDynamoDBException {
+
+        List<Shard> allShards = new ArrayList<>();
+        String streamArn = request.getStreamArn();
+        String lastShardId = request.getExclusiveStartShardId();
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("cache miss in describe stream cache for key: {}", streamArn);
+        }
+        // TODO metrics: count cache miss
+
+        DescribeStreamResult result = null;
+        DescribeStreamResult previousResult;
+        do {
+            request = new DescribeStreamRequest().withStreamArn(streamArn).withExclusiveStartShardId(lastShardId);
+            previousResult = result;
+            result = super.describeStream(request);
+
+            if (result != null && result.getStreamDescription().getShards() != null) {
+                List<Shard> shards = result.getStreamDescription().getShards();
+                allShards.addAll(shards);
+                lastShardId = result.getStreamDescription().getLastEvaluatedShardId();
+            } else {
+                // since the request was null, update lastShardId to null since we can't get the next request
+                // from a null response
+                lastShardId = null;
+            }
+        } while (lastShardId != null);
+
+        if (result != null) {
+            StreamDescription streamDescription = result.getStreamDescription();
+            streamDescription.setShards(allShards);
+            result.setStreamDescription(streamDescription);
+        } else if (previousResult != null) {
+            StreamDescription streamDescription = previousResult.getStreamDescription();
+            streamDescription.setShards(allShards);
+            result = previousResult;
+            result.setStreamDescription(streamDescription);
+        } else {
+            result = new DescribeStreamResult().withStreamDescription(new StreamDescription().withStreamArn(streamArn)
```

Comment:
Wasn't sure if we want to return null on null `DescribeStreamResult` responses.  Right now we don't handle null case responses, probably because we don't expect it from the aws DescribeStream call.  Here I just return a basic stream description result.

Reply:
updated in f3d1d9ef8aec9f343ef093ee040b2fe75c8f169c